jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -2014,79 +2014,80 @@ ecma_op_object_get_enumerable_property_names (ecma_object_t *obj_p, /**< routine
   for (uint32_t i = 0; i < prop_names_p->item_count; i++)
   {
     /* 4.a */
-    JERRY_ASSERT (ecma_is_value_string (names_buffer_p[i]));
-
-    ecma_string_t *key_p = ecma_get_string_from_value (names_buffer_p[i]);
-
-    /* 4.a.i */
-    ecma_property_descriptor_t prop_desc;
-    ecma_value_t status;
-
-#if ENABLED (JERRY_BUILTIN_PROXY)
-    if (ECMA_OBJECT_IS_PROXY (obj_p))
+    if (ecma_is_value_string (names_buffer_p[i]))
     {
-      status = ecma_proxy_object_get_own_property_descriptor (obj_p, key_p, &prop_desc);
-    }
-    else
-#endif /* ENABLED (JERRY_BUILTIN_PROXY) */
-    {
-      status = ecma_op_object_get_own_property_descriptor (obj_p, key_p, &prop_desc);
-    }
+      ecma_string_t *key_p = ecma_get_string_from_value (names_buffer_p[i]);
 
-    if (ECMA_IS_VALUE_ERROR (status))
-    {
-      ecma_collection_free (prop_names_p);
-      ecma_collection_free (properties_p);
+      /* 4.a.i */
+      ecma_property_descriptor_t prop_desc;
+      ecma_value_t status;
 
-      return NULL;
-    }
-
-    /* 4.a.ii */
-    if ((prop_desc.flags & ECMA_PROP_IS_ENUMERABLE) != 0)
-    {
-      /* 4.a.ii.1 */
-      if (option == ECMA_ENUMERABLE_PROPERTY_KEYS)
+  #if ENABLED (JERRY_BUILTIN_PROXY)
+      if (ECMA_OBJECT_IS_PROXY (obj_p))
       {
-        ecma_collection_push_back (properties_p, ecma_copy_value (names_buffer_p[i]));
+        status = ecma_proxy_object_get_own_property_descriptor (obj_p, key_p, &prop_desc);
       }
       else
+  #endif /* ENABLED (JERRY_BUILTIN_PROXY) */
       {
-        /* 4.a.ii.2.a */
-        ecma_value_t value = ecma_op_object_get (obj_p, key_p);
+        status = ecma_op_object_get_own_property_descriptor (obj_p, key_p, &prop_desc);
+      }
```

Comment:
Do we really need the `if-else` here? The `ecma_op_object_get_own_property_descriptor` already checks internally if the input object is a Proxy or not. For now maybe we could use the `ecma_op_object_get_own_property_descriptor` in every case.